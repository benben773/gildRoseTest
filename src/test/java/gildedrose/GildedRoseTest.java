package gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class GildedRoseTest {

    @Test
    public void approveTest() {

        CombinationApprovals.verifyAllCombinations(
                this::getString,
                new String[]{"foo","Aged Brie","Sulfuras, Hand of Ragnaros","Backstage passes to a TAFKAL80ETC concert"},
                new Integer[]{-1,0,1,10,11,12},
                new Integer[]{-1,0,1,49,50,51}
        );
    }

    private String getString(String foo, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(foo, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }

}
