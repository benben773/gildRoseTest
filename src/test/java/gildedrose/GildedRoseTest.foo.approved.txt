[Lgildedrose.Item;@27f8302d